### ISSUE TITLE
Picotable 'Select All' doesn't respect filters was made previously

### ISSUE THREAD

**Issue by wardal** [shuup/shuup#2056]
Use case: 
we have some list of instances in sa, for example orders or products, etc. 
I use filters to filter orders somehow or filter by name. Then, when I click Select All in dropdown I see message 'All %n Items selected', and it uses correct number of instances I have after filtering. But if I will do any mass operation (for example Orcer Cancel) it will use all orders, and not orders I have after filtering.

Why? :

```
                case "select_all":
                    ctrl.selectAllProducts();
                    const totalItemCount = ctrl.vm.data().pagination.nItems;
                    const selectAllMessage = interpolate(gettext("All %s Items Selected"), [totalItemCount]);
                    window.Messages.enqueue({ tags: "info", text: selectAllMessage });
                    return;
```

And in selectAllProducts it change allItemsSelected prop to true, when default is false:

```
ctrl.selectAllProducts = function () {
            ctrl.selectAllListedProducts();
            ctrl.vm.allItemsSelected(true);
        };
```

Then in payload it check this prop and use 'all' string if it's true or ids of instances that was selected (btwn, we have correct array of selected items after filtering in originalValues, but as property was true, it uses 'all' string  ):
```
var payload = {
                "action": value,
                "values": (ctrl.vm.allItemsSelected() ? "all" : originalValues)
            };
```

As I used example with Order Cancel, checking what inside:

```
class CancelOrderAction(PicotableMassAction):
    label = _("Cancel")
    identifier = "mass_action_order_cancel"

    def process(self, request, ids):
        shop = get_shop(request)
        if isinstance(ids, six.string_types) and ids == "all":
            query = Q(shop=shop)
        else:
            query = Q(id__in=ids, shop=shop)
        for order in Order.objects.filter(query):
            if not order.can_set_canceled():
                continue
            order.set_canceled()
```

On back-end it see 'all' string and use only shop to filter Orders. Our selected orders was avoid.

Main problem now in allItemsSelected property, as after filtering originalValues stores correct array of selected items, but as property was set to true, it uses 'all' in payload.

What I think now:
1) Avoid using this property and sent what was selected. But array could be very big, if there, for example, 2000 items was selected.
2) Use 'all' as before, but sent also filter args and make filtering on back-end also.
3) Check somehow that filtering was applied, and in such case send not 'all' by array of ids. Also could be big POST if there a lot of items left after filtering.
4) Your proposals?

**Comment by wardal** [comment]
Ping @tulimaki @chessbr @maximlion @Pikkupomo to make some noise around this :)

**Comment by chessbr** [comment]
@wardal Yes, looks a nice feature. What you can do is to add a new option in Shuup like: "Select All Filtered items". Then you get the filters from the picotable (`window.picotable.ctrl.getFilters()`) and pass it to POST data. Then, instead of having a `all` key you could use a `filtered` and expect to also receive the filters form the request.

Does this make sense?

**Comment by wardal** [comment]
@chessbr yes, I understood

**Comment by wardal** [comment]
Hey @chessbr 
I started to work on this task and now I'm not sure our solution with adding new 'filtered' case in picotable and sending filter params in POST data is best thing what we can do here.

As all list pages in /sa/ has their own set of filter params and mass actions, we needed to change every mass action view for /sa/ to handle filtered in POST data. This seems required much more time I expected before.

On current state I prefer to create new mass action 'Select Filtered' like you proposed and just send id's in POST data without string 'filtered'. In this case we can leave all mass actions views and they will work, as already written to work will 'all' string or ids in POST data.

**Comment by maximlion** [comment]
Hello @wardal, are you still working on this?

**Comment by wardal** [comment]
Hey @maximlion , yes, but please read whole thread, cuz I have blocker and ask for proposals in previous msg.
